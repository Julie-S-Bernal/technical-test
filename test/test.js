const request = require('supertest');

const paths = ['/about-page', '/jobs' , '/valves'];

describe('Project Test:', () => {
    describe('Page Ping:', () => {
        paths.map( (path) => {
            it('should respond with 200', (done) => {
                request(require('../app.js'))
                .get(path)
                .expect(200, (err, res) =>  {
                    done();
                });
            });
        });
        it('Should respond 404', (done) => {
            request(require('../app.js'))
            .get('/wrong-url')
            .expect(200, (err, res) =>  {
                done();
            });
        }); 
    });
    describe('Ensure page content', () => {
        paths.map( (path) => {
            it('should match snapshots 200', (done) => {
                request(require('../app.js'))
                .get(path)
                .expect(200, (err, res) =>  {
                    expect(res.text).toMatchSnapshot();
                    done();
                });
            });
        });    
    });
});