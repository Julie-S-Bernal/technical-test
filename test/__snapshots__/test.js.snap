// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Test: Ensure page content should match snapshots 200 1`] = `
"<!doctype html>
<html>
    <head>
        <title>Welcome to Acme</title>
    </head>
    <body>
      <h1 id=\\"thisistheaboutpage\\">This is the About page</h1>
<p>Acme Co. is a reputable maker of widgets and is an international brand.</p>
<p>Thank you for your interest in our services. Please contact us at enquiries@acme.com.</p>
    </body>
</html>
"
`;

exports[`Project Test: Ensure page content should match snapshots 200 2`] = `
"<!doctype html>
<html>
    <head>
        <title>Welcome to Acme</title>
    </head>
    <body>
      <h1 id=\\"jobsatacmeco\\">Jobs at Acme Co.</h1>
<p>Acme Co. is often seeking candidates in the areas of marketing, finance, customer support and sales. If you are an enthusiastic go-getter, you don't need to look any further for your next step up the corporate career ladder.</p>
<p>Send us an email at careers@acme.com for more information.</p>
    </body>
</html>
"
`;

exports[`Project Test: Ensure page content should match snapshots 200 3`] = `
"<!doctype html>
<html>
    <head>
        <title>Welcome to Acme</title>
    </head>
    <body>
      <h1 id=\\"valves\\">Valves</h1>
<p>Acme Co. valves are amongst the highest quality in the industry. Whether it's for industrial, commercial, medical or space exploration, you can always count on an Acme Co. valve.</p>
    </body>
</html>
"
`;
